### PR TITLE
Add reductions example

### PR DESCRIPTION
**Pull request by awolant**
Signed-off-by: Albert Wolant <awolant@nvidia.com>

#### Why we need this PR?
- It adds new example for reductions

#### What happened in this PR?
 - Documentation (including examples):
     *New notebook*

**JIRA TASK**: *[Use DALI-1691]*

### CODE REVIEW

**Comment by awolant** [comment]
!build

**Comment by JanuszL** [comment]
I would add an example with GPU variant of these operators.

**Comment by awolant** [comment]
> I would add an example with GPU variant of these operators.

Done

**Comment by awolant** [comment]
!build

File: docs/examples/general/reductions.ipynb [hunk 1]
```diff
@@ -23,7 +23,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 3,
+   "execution_count": 17,
```
Comment (JanuszL) [inline docs/examples/general/reductions.ipynb:26]: Please fix the execution count.

**Review by JanuszL — APPROVED** [review]

File: docs/examples/general/reductions.ipynb [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+    "    variance = fn.reductions.variance(input, mean)\n",
+    "    \n",
+    "    pipe.set_outputs(mean, std_dev, variance)\n",
+    "\n",
+    "run_and_print(pipe)"
+   ]
+  },
+  {
+   "source": [
+    "In the code sample above applying reductions resulted in changing the output type. To provide desired output type `dtype` argument can be used."
```
Comment (jantonguirao) [inline docs/examples/general/reductions.ipynb:368]:
```suggestion
    "In the code sample above, applying reductions resulted in changing the output type. The argument `dtype` can be used to specify the desired output data type."
```

File: docs/examples/general/reductions.ipynb [hunk 3]
```diff
@@ -0,0 +1,532 @@
...
+    "    sum_float = fn.reductions.sum(input, dtype=types.FLOAT) \n",
+    "    \n",
+    "    pipe.set_outputs(sum_int_64, sum_float)\n",
+    "\n",
+    "run_and_print(pipe)"
+   ]
+  },
+  {
+   "source": [
+    "Not all types combinations are supported. Default behaviour of operators varies. General rule is for the output type to be able to accommodate the result depending on the input type. For example for input type `INT32` default output type of sum is `INT32` and default output type of mean is `FLOAT`.\n",
```
Comment (jantonguirao) [inline docs/examples/general/reductions.ipynb:400]:
```suggestion
    "Not all data types combinations are supported. The default behaviour varies from operator to operator. A general rule is for the output type to be able to accommodate the result depending on the input type. For example, for an input type `INT32` the default output type of sum is `INT32` and the default output type of mean is `FLOAT`.\n",
```

File: docs/examples/general/reductions.ipynb [hunk 4]
```diff
@@ -0,0 +1,532 @@
...
+    "\n",
+    "All reductions can be offloaded to the GPU. GPU variants work the same way as their CPU counterparts. Below we show a code sample containing all reductions offloaded to the GPU with various parameters."
```
Comment (jantonguirao) [inline docs/examples/general/reductions.ipynb:402]:
```suggestion
    "All reductions can be offloaded to the GPU. The GPU variants work the same way as their CPU counterparts. Below we show a code sample containing all reductions offloaded to the GPU with various parameters."
```

**Review by jantonguirao — APPROVED** [review]

**Comment by awolant** [comment]
!build

**Pull request merged**